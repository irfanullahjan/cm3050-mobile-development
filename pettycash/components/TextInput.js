import { useTheme } from "@react-navigation/native";
import { useField } from "formik";
import { useState } from "react";
import {
  StyleSheet,
  Text,
  TextInput as TextInputReactNative,
  View,
} from "react-native";

export function TextInput({ name, placeholder, ...props }) {
  const [field, meta, helpers] = useField(name);
  const [focused, setFocused] = useState(false);
  const { colors } = useTheme();

  return (
    <View style={styles.container}>
      <Text style={[styles.label, { color: colors.text }]}>{placeholder}</Text>
      <TextInputReactNative
        style={[
          styles.input,
          {
            color: colors.text,
            borderColor: focused ? colors.primary : colors.border,
            ...(focused ? {backgroundColor: colors.card} : {}),
          },
        ]}
        {...props}
        onChangeText={helpers.setValue}
        onBlur={() => {
          helpers.setTouched(true);
          setFocused(false);
        }}
        onFocus={() => setFocused(true)}
        value={field.value}
      />
      {meta.touched && meta.error ? (
        <Text style={{ color: colors.notification }}>{meta.error}</Text>
      ) : (
        <Text> </Text>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    marginBottom: 15,
  },
  label: {
    fontSize: 16,
  },
  input: {
    fontSize: 16,
    borderWidth: 1,
    padding: 10,
    marginVertical: 5,
  },
});
