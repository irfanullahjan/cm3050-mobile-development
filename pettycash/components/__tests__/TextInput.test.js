import { useField } from "formik";
import React from "react";
import { render } from "react-native-testing-library";

import { TextInput } from "../TextInput";

jest.mock("formik");

describe("TextInput", () => {
  it("renders correctly", () => {
    useField.mockReturnValue([
      { value: "test" },
      { error: "test", touched: true },
      { setValue: jest.fn(), setTouched: jest.fn() },
    ]);
    const tree = render(<TextInput name="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
