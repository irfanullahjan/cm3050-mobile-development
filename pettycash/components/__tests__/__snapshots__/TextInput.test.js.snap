// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  />
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 1,
          "fontSize": 16,
          "marginVertical": 5,
          "padding": 10,
        },
        Object {
          "borderColor": "rgb(216, 216, 216)",
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
    value=""
  />
  <Text>
     
  </Text>
</View>
`;

exports[`TextInput renders correctly with label 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    Test label
  </Text>
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "borderWidth": 1,
          "fontSize": 16,
          "marginVertical": 5,
          "padding": 10,
        },
        Object {
          "borderColor": "rgb(216, 216, 216)",
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
    value="Input value"
  />
  <Text>
     
  </Text>
</View>
`;

exports[`TextInput renders correctly with placeholder 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  />
  <TextInput
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    placeholder="Test placeholder"
    style={
      Array [
        Object {
          "borderWidth": 1,
          "fontSize": 16,
          "marginVertical": 5,
          "padding": 10,
        },
        Object {
          "borderColor": "rgb(255, 59, 48)",
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
    value="test"
  />
  <Text
    style={
      Object {
        "color": "rgb(255, 59, 48)",
      }
    }
  >
    test
  </Text>
</View>
`;
